LogIn/LoginViewController.swift

```diff
@@ -45,48 +67,89 @@ class LoginViewController: UIViewController {
         button.isEnabled = false
         return button
     }()
-
+    
+    init(model: LoginViewModelType) {
+        self.model = model
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         
         setupViews()
+        bind(to: model)
+    }
+    
+    @objc
+    private func emailTextEntered() {
+        if let text = emailTextField.text {
```

Comment:
nil checking and other transformations  must be done in the viewModel

Reply:
done